### DISCUSSION TITLE
[Question] Can I use redis/database to manage session?

### DISCUSSION THREAD

**Discussion by luozhouyang** [flet-dev/flet#2787] [Q&A]
I want to deploy the flet app in a k8s cluster, with many pods. So how flet manage the user sessions? Can I use redis or a database to customize sessions management?

**Comment by 50Bytes-dev** [comment]
I have the same question. I've been looking at the Flet code and it looks like it won't work. It seems that we need to make an implementation to support custom session managers

**Comment by 50Bytes-dev** [comment]
@luozhouyang Have you made any progress on this?

**Comment by 50Bytes-dev** [comment]
@FeodorFitsner, can you tell me what I can do? Should I make support for a custom session manager?

**Reply by FeodorFitsner** [reply]
It would be great to have a pluggable session storage, of course. It's, actually, not a session storage, but page "snapshot" in this class: https://github.com/flet-dev/flet/blob/main/sdk/python/packages/flet/src/flet/fastapi/flet_app.py

**Reply by 50Bytes-dev** [reply]
I need to improve the FletAppManager. Am I right?

**Reply by FeodorFitsner** [reply]
I think so, sessions are stored in app manager and on session re-connect, if it's a cluster of web nodes, every node should be able to pickup session.

**Reply by 50Bytes-dev** [reply]
Is it even possible to save this in Redis? Any idea how to scale a Flet application at all?

**Reply by 50Bytes-dev** [reply]
What about the events that are used on the page?

**Reply by 50Bytes-dev** [reply]
@FeodorFitsner, please take a look at this. I think it's what we need.
https://github.com/uqfoundation/dill

**Reply by FeodorFitsner** [reply]
I don't think you need that to persist controls. It's just a dictionary of string dictionary: https://github.com/flet-dev/flet/blob/main/sdk/python/packages/flet-core/src/flet_core/page.py#L1200 - it could be nicely mapped to Redis's HSET.

**Comment by 50Bytes-dev** [comment]
@ndonkoHenri can you help on this issue?
